R/integrateICT.R

```diff
@@ -0,0 +1,137 @@
+  ictScen <- config[scen, "techScen"] %>%
+    unlist() %>%
+    (\(x) strsplit(x, "_")[[1]])() %>%
+    as.list() %>%
+    setNames(c("ssp", "estimate"))
+
+  # lower temporal boundary
+  periodBegin <- config[scen, "periodBegin"] %>%
+    unlist()
+
+  # upper temporal boundary of historical data
+  endOfHistory <- config[scen, "endOfHistory"] %>%
+    unlist()
+
+
+
+  # PROCESS DATA ---------------------------------------------------------------
+
+  ## Extrapolate historical ICT demand and correct appliances_light fe demand ====
+
+  if (isFALSE(expandProjections)) {
+
+    # Extract AL electricity demand
+    appliancesElec <- fe %>%
+      filter(.data$enduse == "appliances_light", .data$carrier == "elec")
+
+    # Filter ICT data and calculate historical shares
+    ictDemand <- ict %>%
+      filter(.data$scenario %in% c("history", ictScen$ssp),
+             .data$variable == ictScen$estimate)
+
+    # Calculate historical ICT/appliances share
+    ictShares <- ictDemand %>%
+      filter(.data$scenario == "history") %>%
+      select(.data$region, .data$period, .data$value) %>%
+      right_join(appliancesElec, by = c("region", "period"), suffix = c("ICT", "App")) %>%
+      mutate(share = .data$valueICT / .data$valueApp) %>%
+      select("region", "period", "share") %>%
+
+      # extrapolate missing entries w/ constant historical share
+      interpolate_missing_periods(value = "share", expand.values = TRUE)
+
+
+    # Fill historical ICT demands and combine with future scenarios
+    ictComplete <- ictDemand %>%
+      filter(.data$scenario == "history") %>%
+      interpolate_missing_periods(periodBegin:endOfHistory) %>%
+      left_join(ictShares, by = c("region", "period")) %>%
+      left_join(appliancesElec %>%
+                  select("region", "period", "value"),
+                by = c("region", "period"),
+                suffix = c("ICT", "Total")) %>%
+      mutate(value = ifelse(!is.na(.data$valueICT), .data$valueICT, .data$valueTotal * .data$share),
+             .keep = "unused") %>%
+      rbind(ictDemand %>%
+              filter(.data$scenario != "history") %>%
+              mutate(scenario = scen)) %>%
+      mutate(variable = "ict.elec|fe") %>%
+      select(-"enduse", -"carrier") %>%
+      as.quitte() %>%
+      missingToNA()
+
+    # Subtract ICT from AL electricity demand
+    appliancesElec <- appliancesElec %>%
+      left_join(ictComplete %>% select(.data$region, .data$period, .data$value),
+                by = c("region", "period"),
+                suffix = c("", "ICT")) %>%
+      mutate(value = .data$value - .data$valueICT, .keep = "unused")
+
+    # Rebuild final energy data without ICT in appliances
+    fe <- fe %>%
+      filter(!(.data$enduse == "appliances_light" & .data$carrier == "elec")) %>%
+      rbind(appliancesElec)
+
+
+    return(list(fe = fe,
+                feICT = ictComplete))
+
+  } else {
+
+    ## Extrapolate future ICT demand with constant 2050 demand ====
```

Comment:
I was irritated to not find 2050 explicitly in the code. Tiny change to make this a bit clearer perhaps.
```suggestion
    ## Extrapolate future ICT demand constantly ====

    # original projections run only until 2050 
```